pkg/provider/branch/provider.go

```diff
@@ -462,13 +462,35 @@ func (b *branchENIProvider) GetPool(_ string) (pool.Pool, bool) {
 // IsInstanceSupported returns true for linux node as pod eni is only supported for linux worker node
 func (b *branchENIProvider) IsInstanceSupported(instance ec2.EC2Instance) bool {
 	limits, found := vpc.Limits[instance.Type()]
-	if !found {
-		return false
-	}
-	if instance.Os() == config.OSLinux && limits.IsTrunkingCompatible {
-		return true
+	supported := found && instance.Os() == config.OSLinux && limits.IsTrunkingCompatible
+
+	// 1, since VPC CNI will not send events for nodes that are labelled as not-supported
+	// 2, since VPC CNI will not keep sending events if nodes have been labelled as true and trunk has been attached
+	// here calling GetNode API will not cause large call volume
+	if node, err := b.apiWrapper.K8sAPI.GetNode(instance.Name()); err != nil {
+		b.log.Error(err, "couldn't get the node when checking if the node is supported for trunk interface", "NodeName", instance.Name())
+	} else {
+		if node.Labels != nil {
+			if _, ok := node.Labels[config.HasTrunkAttachedLabel]; ok {
+				labelValue := config.BooleanTrue
+				if !supported {
+					labelValue = config.NotSupportedEc2Type
+				}
+				updated, err := b.apiWrapper.K8sAPI.AddLabelToManageNode(node, config.HasTrunkAttachedLabel, labelValue)
```

Comment:
What is the scenario where we the AddLabelToManageNode return `nil` but the label is `not updated`?

Reply:
when the label key/value pair is same as requested, the update failed but error is nil
```
if node.Labels[labelKey] == labelValue {
		return false, nil
	}
```